### PR TITLE
Feature/file config

### PR DESCRIPTION
**Pull request by Ofir-Lapid**
## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: kubebeat/beater/file_system_fetcher.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+type FileSystemFetcher struct {
+	filesPaths []string // Files and directories paths for the fetcher to extract info from
+}
+
+const (
+	FILE_SYSTEM_INPUT_TYPE = "file-system"
+)
+
+type FileSystemResourceData struct {
```
Comment (Ofir-Lapid) [inline kubebeat/beater/file_system_fetcher.go:19]: Should I add a "struct" general description ? What's the purpose .... ?

File: kubebeat/kubebeat.yml [hunk 2]
```diff
@@ -80,8 +94,8 @@ output.elasticsearch:
   #username: "elastic"
   #password: "changeme"
 
-# ------------------------------ Logstash Output -------------------------------
-#output.logstash:
+  # ------------------------------ Logstash Output -------------------------------
```
Comment (yashtewari) [inline kubebeat/kubebeat.yml:97]: Looks like a bunch of changes in which spaces have been prefixed to lines? Not sure what that is.

**Review by yashtewari — APPROVED** [review]

File: kubebeat/beater/file_system_fetcher.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		results = append(results, result)
+	}
+
+	return results, nil
+}
+
+func (f *FileSystemFetcher) Stop() {
+}
+
+func FromFileInfo(info os.FileInfo, path string) FileSystemResourceData {
```
[the remark this answers is not in the document]
Comment (ofiriro3) [reply kubebeat/beater/file_system_fetcher.go:56]: I am not sure about it, my idea was to create a parser for
FileInfo -> FileSystemResourceData

I am not sure who will use it, but parser typically are public

File: kubebeat/beater/file_system_fetcher.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+	if info == nil {
+		return FileSystemResourceData{}
+	}
+
+	stat := info.Sys().(*syscall.Stat_t)
```
[the remark this answers is not in the document]
Comment (ofiriro3) [reply kubebeat/beater/file_system_fetcher.go:62]: Not sure what should I explain here

The function is a parser it takes a FileInfo and parse it to FileSystemResourceData

File: kubebeat/beater/file_system_fetcher.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	uid := stat.Uid
+	gid := stat.Gid
+	u := strconv.FormatUint(uint64(uid), 10)
+	g := strconv.FormatUint(uint64(gid), 10)
+	usr, _ := user.LookupId(u)
+	group, _ := user.LookupGroupId(g)
+	mod := strconv.FormatUint(uint64(info.Mode().Perm()), 8)
+
+	data :=
+		FileSystemResourceData{
```
[the remark this answers is not in the document]
Comment (ofiriro3) [reply kubebeat/beater/file_system_fetcher.go:72]: Changed, my linter also approved skipping one line - I guess both are ok.
Anycase, I changed it

File: kubebeat/kubebeat.yml (hunk 2 above)
Comment (ofiriro3) [reply kubebeat/kubebeat.yml:97]: I'll fix it in a later stage

File: kubebeat/kubebeat.yml [hunk 6]
```diff
@@ -5,6 +5,20 @@
 kubebeat:
   # Defines how often an event is sent to the output
   period: 1s
+  files: [
+      "/hostfs/etc/kubernetes/scheduler.conf",
```
[the remark this answers is not in the document]
Comment (ofiriro3) [reply kubebeat/kubebeat.yml:9]: Nice idea, but in my opinion, it should be as concrete as we can (unless we will have 100 in one dir).
If you feel it's important we can discuss on it later on

File: kubebeat/kubebeat.yml (hunk 2 above)
Comment (yashtewari) [reply kubebeat/kubebeat.yml:97]: Nevermind, saw later that this file is just for POC usage. It won't be used when we integrate this code upstream.

**Pull request merged**